pennylane/operation.py

```diff
@@ -815,14 +964,16 @@ def inv(self):
             self.inverse = not self._inverse
         return self
 
-    @property
-    def matrix(self):
-        op_matrix = self._matrix(*self.parameters)
+    def matrix(self, wire_order=None):
+        canonical_matrix = self.compute_matrix(*self.parameters, **self.hyperparameters)
 
         if self.inverse:
-            return qml.math.conj(qml.math.T(op_matrix))
+            canonical_matrix = qml.math.conj(qml.math.T(canonical_matrix))
```

Comment:
I hope this will disappear with the new inverse system. It is annoying to replicate it when overwriting the method, and I am very confused which operator has to replicate it (just following the status quo below).

Reply:
🤔

Reply:
I can't stop thinking about this.

In the matrix case, we have `compute_matrix` (static, designed to be overridden) and `matrix` (instance method, contains post-processing logic).

In the generator case, we have `generator` (instance method, designed to be overridden), and `utils.get_generator(op)` (external function to add post-processing logic).

Different solutions for the same problem: how do we provide a stable API for developers to add a representation, while also abstracting away how that representation changes with the operator.

Reply:
Still need to work this one out conclusively!!